### DISCUSSION TITLE
code callouts? (via asciidoc?)

### DISCUSSION THREAD

**Discussion by colemickens** [thuliteio/doks#129] [Q&A]
Hi!

First, let me say Thank You! `doks` looks really great!

I have a killer feature that I need for my docs -- code callouts. Is this possible to achieve with `hugo` and `doks`? I know that it's possible to get `hugo` to use `asciidoc` to generate HTML, but I'm not sure how that approach would work out with `doks`.

Any suggestions? I'd be fine with:
* using asciidoc with `doks`
* achieveingcode callouts via some other mechanism (some hugo plugin to process the source code and then extract out the callouts into html that is friendly to `doks`, I'm not sure)

I think ideally, though, I could still use asciidoc, as it has some other callout/aside type features that markdown lacks.

Here's an example of what I mean by code callouts: 
![code-callout-ex](https://user-images.githubusercontent.com/327028/107305725-58797600-6a38-11eb-86e0-7932bcc37bb2.PNG)

I'll probably just try and see soon, but I'm curious what others think. thanks!

**Comment by epogrebnyak** [comment]
>  I know that it's possible to get hugo to use asciidoc to generate HTML

Any source about this? I think Hugo renders just markdown, but not asciidoc.

**Reply by colemickens** [reply]
The example shown is actually from an article about it. https://blog.arkey.fr/2020/04/23/tackling-hugo-integration-of-asciidoctor/

That article shows how to modify hugo's built-in support for asciidoc, etc. There are many other articles on it, most of them are linked from that one I just linked.

**Comment by epogrebnyak** [comment]
Thanks! It may be quite a project to provide asciidoc functionality out of
the box, let's see what others may say about it.

Meanwhile, an imperfect substitute - maybe Hugo own codeblock with line
numbers? There are no callouts to line numbers but you can refer to them in
markdown text.

вт, 9 февр. 2021 г., 9:32 Cole Mickens <notifications@github.com>:
